shell/artifacts/Social/source/EditPost.js

```diff
@@ -35,25 +40,29 @@ defineParticle(({DomParticle, html, log}) => {
   right: 8px;
   top: 8px;
 }
-[${host}] [post-buttons] i {
+[${host}] [post-buttons] > * {
   border-radius: 100%;
   color: lightgrey;
   display: inline-block;
   padding: 8px;
 }
-[${host}] [post-buttons] i:active {
+[${host}] [post-buttons] > *:active {
   background-color: #b0e3ff;
 }
-[${host}] [post-buttons] .button-live {
+[${host}] [post-buttons] > [active] {
+  color: black;
+}
+[${host}] [post-buttons] > [active][primary] {
   color: green;
 }
 </style>
 <div ${host}>
   <div post-buttons>
     <i class="material-icons" on-click="onDeletePost">delete</i>
-    <i class="material-icons" on-click="onAttachPhoto">attach_file</i>
-    <i class="{{saveClasses}}" on-click="onSavePost">done</i>
+    <firebase-upload active accept="image/*" on-upload="onAttachPhoto"><i class="material-icons">attach_file</i></firebase-upload>
+    <i class="material-icons" primary active$="{{saveButtonActive}}" on-click="onSavePost">done</i>
   </div>
+  <img src="{{imageUrl}}">
```

Comment:
Ideally we'd have the image width/height to incorporate here, that's beyond MVP but I'm curious how you would go about getting it?

Reply:
I don't have a good way off the top of my head to get that information. Will have to think about it.

Reply:
Another reason to want size is to improve layout performance through ahead-of-time sizing when possible. But maybe that's gotten fancier and more efficient somehow in recent days. We're far from that being a big issue.